dashboard/client/src/pages/serve/ServeApplicationDetailPage.tsx

```diff
@@ -0,0 +1,239 @@
+import {
+  Box,
+  createStyles,
+  InputAdornment,
+  makeStyles,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TextField,
+  TextFieldProps,
+  Typography,
+} from "@material-ui/core";
+import { Autocomplete, Pagination } from "@material-ui/lab";
+import dayjs from "dayjs";
+import React, { ReactElement } from "react";
+import { useParams } from "react-router-dom";
+import { CodeDialogButton } from "../../common/CodeDialogButton";
+import { CollapsibleSection } from "../../common/CollapsibleSection";
+import { DurationText } from "../../common/DurationText";
+import { MetadataSection } from "../../components/MetadataSection";
+import { StatusChip } from "../../components/StatusChip";
+import { HelpInfo } from "../../components/Tooltip";
+import { MainNavPageInfo } from "../layout/mainNavContext";
+import { useServeApplicationDetails } from "./hook/useServeApplications";
+import { ServeDeploymentRow } from "./ServeDeploymentRow";
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    table: {
+      tableLayout: "fixed",
+    },
+    helpInfo: {
+      marginLeft: theme.spacing(1),
+    },
+  }),
+);
+
+const columns: { label: string; helpInfo?: ReactElement; width?: string }[] = [
+  { label: "Name" },
+  { label: "Status" },
```

Comment:
```suggestion
  { label: "Status message" },
```

[Nit] `"Message"` feels a bit vague on the UI.